### PR TITLE
temporary mobile endpoint for ICDS release

### PR DESCRIPTION
**Pull request by sravfeyn**
@nickpell This is a subset of [heartbeat spec](https://docs.google.com/document/d/1RhdJPD6ZvRGcUsSgULMt05yXUQG6j_FLCAi1fMMEtyQ/edit#) that Aliza requested as a high priority for ICDS release

@amstone326 FYI

### CODE REVIEW

**Review by nickpell — APPROVED** [review]

File: corehq/apps/app_manager/models.py [hunk 1]
```diff
@@ -5001,6 +5001,10 @@ def key_server_url(self):
         return reverse('key_server_url', args=[self.domain])
 
     @absolute_url_property
+    def heartbeat_url(self):
```
Comment (snopoke) [inline corehq/apps/app_manager/models.py]: where does this get called?

File: corehq/apps/app_manager/models.py [hunk 2]
```diff
@@ -5001,6 +5001,10 @@ def key_server_url(self):
...
+        return reverse('key_server_url', args=[self.domain])
```
Comment (snopoke) [inline corehq/apps/app_manager/models.py]: how come this is a different URL from the `apk_heartbeat_url`?

Comment (sravfeyn) [inline corehq/apps/app_manager/models.py]: Sorry, I was gonna use this in the below code but didn't. I will remove it

Comment (sravfeyn) [inline corehq/apps/app_manager/models.py]: Updated

File: corehq/apps/app_manager/models.py (hunk 1 above)
Comment (sravfeyn) [inline corehq/apps/app_manager/models.py]: Fixed

File: corehq/apps/ota/views.py [hunk 3]
```diff
@@ -361,4 +361,9 @@ def form_valid(self):
 @login_or_digest_or_basic_or_apikey()
 @require_GET
 def heartbeat(request, domain, id):
-    return JsonResponse({})
+    # mobile needs this. This needs to be revisited to actually work dynamically (Sravan June 7, 17)
+    return JsonResponse({
+        "app_id": "value_copied_from_app_id_url_param",
```
Comment (amstone326) [inline corehq/apps/ota/views.py]: Hey @sravfeyn, sorry I just realized this was very unclear, but I meant that this value should be copied from the app_id url param that mobile will include.

**Comment by amstone326** [comment]
@nickpell Please don't pull this yet, have a couple of things that need to be fixed

**Comment by amstone326** [comment]
@sravfeyn Thank you for doing this!! Only issue I see is the thing with the value of `app_id`

File: corehq/apps/ota/views.py (hunk 3 above)
Comment (amstone326) [inline corehq/apps/ota/views.py]: You can expect that mobile will always include a url param in this request with key `app_id`

Comment (sravfeyn) [inline corehq/apps/ota/views.py]: Oops my bad

Comment (sravfeyn) [inline corehq/apps/ota/views.py]: I have fixed this. Sorry about it, just to confirm per the spec the `app_id` should come from the last section of URL https://www.commcarehq.org/a/{domain}/phone/heartbeat/{app_id}/. it's not a get URL param, and that's what I changed this to

**Review by snopoke — APPROVED** [review]

**Pull request merged**